src/views/AddTimerView.tsx

```diff
@@ -0,0 +1,206 @@
+import { useContext, useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { TimerContext } from '../context/TimerContext';
+
+import { Button } from '../components/generic/Button';
+import { Panel } from '../components/generic/Panel';
+import { Select } from '../components/generic/Select';
+
+import TimerSettings from '../components/timers/TimerSettings';
+import type { Timer } from '../types/timers';
+import { convertToMs } from '../utils/helpers';
+
+export default function AddTimer() {
+    const navigate = useNavigate();
+    const [selectedTimer, setSelectedTimer] = useState<string>('Stopwatch');
+    const { timers, addTimer, updateTimer } = useContext(TimerContext);
+
+    // timer index from URL
+    const { index } = useParams();
+
+    // state for this timer
+    const [roundsValue, setRoundsValue] = useState(1);
+    const [timerTime, setTimerTime] = useState({ minutes: 0, seconds: 0 });
+    const [workTime, setWorkTime] = useState({ minutes: 0, seconds: 0 });
+    const [restTime, setRestTime] = useState({ minutes: 0, seconds: 0 });
+    const [description, setDescription] = useState('');
+
+    // Timer types for dropdown
+    const timerOptions = [
+        { id: 1, value: 'Stopwatch', label: 'Stopwatch' },
+        { id: 2, value: 'Countdown', label: 'Countdown' },
+        { id: 3, value: 'XY', label: 'XY' },
+        { id: 4, value: 'Tabata', label: 'Tabata' },
+    ];
+
+    // Change timer type when dropdown option selected
+    const handleTimerChange = (value: string | number) => {
+        if (typeof value === 'string') {
+            setSelectedTimer(value);
+        }
+    };
+
+    // Display timer settings by timer type
+    const displayTimer = () => {
```

Comment:
This function isn't doing much apart from the if statement, and ends up making the AddTimer component a bit busier. You could consider moving this logic to a separate component